sympy/solvers/tests/test_solvers.py

```diff
@@ -1141,12 +1143,14 @@ def test__invert():
 
 def test_issue_4463():
     assert solve(-a*x + 2*x*log(x), x) == [exp(a/2)]
-    assert solve(a/x + exp(x/2), x) == [2*LambertW(-a/2)]
+    #assert solve(a/x + exp(x/2), x) gives [2*LambertW(-a/2), 2*LambertW(a/2)]
+    # expected [2*LambertW(-a/2)]
```

Comment:
You have to tell your solver to use what you know so it can make the same decision: that you only want the first solution. (But how do you know that?)

Reply:
@smichr the equation entering `_lambert` is `_x - 2*log(_u) + 2*log(_x)`  and `_u + _a = 0` and _u is coming out to be positive but it can be positive or negative depending upon _a. Now as _u is positive so it is returning both the solution.

Reply:
The basic problem here i am facing that while converting all the lambert solvable equations to this form `F(X, a..f) = a*log(b*X + c) + d*X + f = 0` , while taking log the solutions are being missing.